Documentation/project-docs/writing-tests.md

```diff
@@ -0,0 +1,93 @@
+This document provides an overview of paradigm we are using in our tests.
+
+# RemoteExecutor
```

Comment:
I agree that with it comes a lot of issues, still we don't have another way to isolate mentioned tests (static properties, whatever...). Just a few minutes before I talked to Dan about the issues with UWP and that we are currently stuck. We need to find a way so solve that, but that's another topic...

Maybe @stephentoub wanna add something?

We could definitely put a warning in it so that people understand that there are also downsides using this pattern. Thanks for the input!